langchain/vectorstores/utils.py

```diff
@@ -1,12 +1,21 @@
 """Utility functions for working with vectors and vectorstores."""
 
+from enum import Enum
 from typing import List
 
 import numpy as np
 
 from langchain.math_utils import cosine_similarity
 
 
+class DistanceStrategy(str, Enum):
+    EUCLIDEAN_DISTANCE = "EUCLIDEAN_DISTANCE"
+    MAX_INNER_PRODUCT = "MAX_INNER_PRODUCT"
+    DOT_PRODUCT = "DOT_PRODUCT"
```

Comment:
Note we keep DOT_PRODUCT for the singlestoredb even if most implentations the inner product, and having both is redundant..

Reply:
what do you mean by "singlestoredb"?